language-support/ts/daml-types/index.ts

```diff
@@ -419,4 +420,82 @@ export const TextMap = <T>(t: Serializable<T>): Serializable<TextMap<T>> => ({
   }
 });
 
-// TODO(MH): `Map` type.
+/**
+ * The counterpart of DAML's `DA.Map.Map K V` type.
+ *
+ * @typeparam K The type of the map keys.
+ * @typeparam V The type of the map values.
+ */
+export interface Map<K, V> {
+  get: (k: K) => V | null;
+  getOr: (k: K, ifNotFound: V) => V;
+  has: (k: K) => boolean;
+  insert: (k: K, v: V) => Map<K, V>;
+  remove: (k: K) => Map<K, V>;
+  keys: () => K[];
+  values: () => V[];
+  kvs: () => [K, V][];
```

Comment:
This is called `entries` in both immutable JS and JS maps and also returns an iterator.